test.gd

```diff
@@ -3,33 +3,189 @@ extends SceneTree
+   duration
+  ])
+  print('Benchmark done')
+  
+func benchmark_obj_rng():
+  print('Benchmarking ...')
+
+  var rng = RandomNumberGenerator.new()
+  var begin = Time.get_unix_time_from_system()
 
-  var duration = 1.0 * OS.get_ticks_msec() - begin
+  for i in range(NUMBER_OF_TESTS):
+    uuid_util.new(rng).free()  # immediately freeing does not seem to add much overhead
 
-  print('uuid/sec: %.02f (time: %sms)' % [ NUMBER_OF_TEST / duration, duration])
+  var duration = 1.0 * Time.get_unix_time_from_system() - begin
+
+  print('uuid/sec: %.02f   avg time: %.4fus   total time: %.2fs' % [
+   NUMBER_OF_TESTS / duration,
+   (duration / NUMBER_OF_TESTS) * 1000000,
+   duration
+  ])
+  print('Benchmark done')
+  
+func benchmark_obj_to_dict():
+  print('Setting up benchmark ...')
+  var uuids = []
+  for i in range(NUMBER_OF_OBJECTS):
+    uuids.append(uuid_util.new())
+
+  print('Benchmarking ...')
+  var begin = Time.get_unix_time_from_system()
+
+  for uuid in uuids:
+    uuid.as_dict()
+
+  var duration = 1.0 * Time.get_unix_time_from_system() - begin
+
+  print('uuid/sec: %.02f   avg time: %.4fus   total time: %.2fs' % [
+   NUMBER_OF_TESTS / duration,
+   (duration / NUMBER_OF_TESTS) * 1000000,
+   duration
+  ])
+  print('Cleaning up ...')
+  for uuid in uuids:
+    uuid.free()
+  print('Benchmark done')
+  
+func benchmark_obj_to_str():
+  print('Setting up benchmark ...')
+  var uuids = []
+  for i in range(NUMBER_OF_OBJECTS):
+    uuids.append(uuid_util.new())
+
+  print('Benchmarking ...')
+  var begin = Time.get_unix_time_from_system()
+
+  for uuid in uuids:
+    uuid.as_string()
+
+  var duration = 1.0 * Time.get_unix_time_from_system() - begin
+
+  print('uuid/sec: %.02f   avg time: %.4fus   total time: %.2fs' % [
+   NUMBER_OF_TESTS / duration,
+   (duration / NUMBER_OF_TESTS) * 1000000,
+   duration
+  ])
+  print('Cleaning up ...')
+  for uuid in uuids:
+    uuid.free()
+  print('Benchmark done')
+  
+func benchmark_comp_raw():
+  print('Setting up benchmark ...')
+  var uuids = []
+  for i in range(NUMBER_OF_TESTS*2):
+    uuids.append(uuid_util.v4())
+  print('Benchmarking ...')
+  var begin = Time.get_unix_time_from_system()
+
+  for i in range(NUMBER_OF_TESTS):
+    var uuid1 = uuids[i*2]
+    var uuid2 = uuids[i*2+1]
+    
+    if uuid1 == uuid2:
+      print("uh-oh")
+
+  var duration = 1.0 * Time.get_unix_time_from_system() - begin
+
+  print('uuid/sec: %.02f   avg time: %.4fus   total time: %.2fs' % [
+   NUMBER_OF_TESTS / duration,
+   (duration / NUMBER_OF_TESTS) * 1000000,
+   duration
+  ])
+  print('Benchmark done')
+  
+func benchmark_comp_obj():
+  print('Setting up benchmark ...')
+  var uuids = []
+  for i in range(NUMBER_OF_OBJECTS*2):
+    uuids.append(uuid_util.new())
+  print('Benchmarking ...')
+  var begin = Time.get_unix_time_from_system()
+
+  for i in range(NUMBER_OF_OBJECTS):
+    var uuid1 = uuids[i*2]
+    var uuid2 = uuids[i*2+1]
+    
+    if uuid1.is_equal(uuid2):
+      print("uh-oh")
```

Comment:
I'd rather see a total number of collision than "uh-oh" written in the console. It should warn the user there is something wrong here.

Reply:
That is not quite the intention of the test, but still a good suggestion. Will implement.